### PR TITLE
Doc ramp expression

### PR DESCRIPTION
**Pull request by IagoLast**
- Add buckets doc
- Add ramp doc
- Prepare colors doc

### CODE REVIEW

File: src/core/style/expressions/buckets.js [hunk 1]
```diff
@@ -4,10 +4,59 @@ import { implicitCast } from './utils';
...
-    /*
-        If input is numeric => args is a breakpoint list
-        If input is categorical => args is a list of category names to map input
-    */
+
+    /**
+     * Given a property create "sub-groups" based on the given breakpoints.
+     * 
+     * 
+     * Imagine a trafic dataset with a speed property. We want to divide the roads in
```
Comment (Jesus89) [inline src/core/style/expressions/buckets.js]: traffic

File: src/core/style/expressions/buckets.js [hunk 2]
```diff
@@ -4,10 +4,59 @@ import { implicitCast } from './utils';
...
+     *  const $procesedSpeed = s.prop('procesedSpeed');
+     *  const style = new carto.Style({
+     *  color: s.ramp(
+     *      s.buckets($procesedSpeed, 'slow', 'medium', 'high'),
+     *      s.palettes.PRISM),
+     * });
+     * ```
+     * 
+     * @param {carto.style.expressions.property} property - The property to be evaluated and interpolated
+     * @param {...carto.style.expressions.expression} breakpoints - Numeric expression containing the different breakpoints.
```
Comment (Jesus89) [inline src/core/style/expressions/buckets.js]: Is this `carto.style.expressions.property` or `carto.style.expressions.prop`.
Why the 3 dots?

Comment (IagoLast) [reply src/core/style/expressions/buckets.js]: both are valid, `s.prop` is an alias for `s.property`.

... dots means you can pass n arguments

File: src/core/style/expressions/palettes.js [hunk 3]
```diff
@@ -1,5 +1,31 @@
...
+ * 
+ * Palettes are constants that allow to use {@link https://carto.com/carto-colors/|cartocolors} easily.
+ * Use them with a {@link carto.style.expressions.ramp|ramp}
+ * 
+ * The following palettes are availiable:
+ *  - Categorical:
+ *      - PRISM
+ *      - EARTH
+ *  - Numeric
+ *      - ...
```
Comment (Jesus89) [inline src/core/style/expressions/palettes.js]: Where are the numeric palettes?

Comment (IagoLast) [reply src/core/style/expressions/palettes.js]: [here](https://github.com/CartoDB/CartoColor/blob/master/cartocolor.js) tagged as cuantitative, names are going to change so I prefer to wait for a decision before copying the full list

Comment (davidmanzanares) [reply src/core/style/expressions/palettes.js]: Please, change this, we may add true numeric palettes in the future (list of numbers for bubble maps), and there are multiple "numeric" palette types I think

**Pull request merged**